### ISSUE TITLE
Perform tests of standalone GMS on clia test box

### ISSUE THREAD

**Issue by malachig** [genome/gms#42]
Perform complete tests of pipeline using the 'clia1' box that has a 1 Tb of ram, 80 processors, 1 Tb of memory, and 2 fusion I/O cards

**Comment by malachig** [comment]
Notes from Matt on using this test box:

I've installed a "bootmail" service that can be configured with a set of email addresses. This service will email a notice at shutdown and startup. This way you can tell when the thing is done imaging itself. Right now a fresh install takes two reboots to get the right kernel in place, install fusion drivers, etc. A re-image looks like this:

root@clia1#  wget -q -O - http://puppetmaster2:3001/precise_image=10.0.70.20
root@clia1#  sudo reboot
receive shutdown email
… time passes as OS install happens …
receive a second shutdown email
receive a startup email

At this point you'd log in and see if it's done imaging:

root@clia1# sudo tail -f /var/log/syslog
… log messages ...
Nov 20 21:56:39 clia1 puppet-agent[25063]: Finished catalog run in 14.61 seconds

You're looking for that "finished catalog run" line.

If you're not sure, ask puppet to run:

root@clia1# /etc/init.d/puppet status
- agent is running
  root@clia1# pkill -HUP puppet

Then watch logs:

root@clia1# sudo tail -f /var/log/syslog

If you want to see what puppet is doing, take a look at the manifest here:

http://git.gsc.wustl.edu/gitweb/?p=puppet-modules.git;a=tree;f=site/systems_clia1_server/manifests;h=eed29a61b7f71a925a0d53f83fcfc31a3d22f2d4;hb=HEAD

Or, in order: init.pp -> packages.pp -> files.pp -> volumes.pp

Again, to re-image:

sudo wget -q -O - http://puppetmaster2:3001/precise_image=10.0.70.20 && sudo /sbin/reboot

It takes several minutes to re-build. I haven't changed the mailto on boot notifications, it's still me, when we see it work a couple times, I'll change it to you all.

If everything went ok you should see:

root@clia1:~# /etc/init.d/puppet status && grep -A1 Failure /var/lib/puppet/state/last_run_report.yaml
- agent is running
        - Failure
        - 0

**Comment by malachig** [comment]
Test procedure:
- log into clia1 (using ldap credentials)
- Change to test user: 'sudo -i -u gmsuser'
- Make sure unwanted remnants of last install are not in /tmp (e.g. /tmp/gms-install-done-local/).  Clean-up if needed
- Make sure unwanted remnants of last install are not in /opt/gms (should be completely empty).  Clean-up if needed.
- cd ~
- git clone https://github.com/genome/gms.git
- cd gms
- make
- log out and log in again!

**Comment by malachig** [comment]
When the clia1 box is re-imaged, here are a few minor tweaks I perform for convenience to allow logging in without entering password each time.

``` bash
ssh mgriffit@clia1
sudo mkdir /gscuser
sudo chown mgriffit:info /gscuser
mkdir /gscuser/mgriffit
cd /gscuser/mgriffit
mkdir .ssh
touch .ssh/authorized_keys
echo "if [ -f ~/.bashrc ]; then" >> .bash_profile
echo "  . ~/.bashrc" >> .bash_profile
echo fi >> .bash_profile
echo "alias ll='ls -l'" >> .bashrc
```

**Comment by malachig** [comment]
The sGMS install on clia1 is currently broken.  In order to get it to work you have to do the following _before_ running the make:

Comment out all lines referencing 'repo.gsc.wustl.edu' in /etc/apt/sources.list.d/.  You may also need to do a 'sudo apt-get update' before running the make, but this is probably not necessary since this should be done for you early on in the make process.

**Comment by malachig** [comment]
There is still something wrong with the package config on clia1 where apt config setup by puppet and the makefile are likely conflicting...

I was eventually able to get the build to proceed successfully using a combination of the following (but I still need to figure out exactly what is needed to reproduce the install consistently).

``` bash
sudo /etc/init.d/puppet status #check status of puppet
sudo /etc/init.d/puppet stop #stop puppet
sudo rm -f /etc/apt/sources.list.d/*  #remove config files created by puppet
#cp sources.list that is used in the vanilla ubuntu install
sudo apt-get update #update apt to use the new apt sources
sudo apt-get -f install #fix broken packages
cd ~
git clone https://github.com/genome/gms.git
cd gms
make #run the gms install
sudo /etc/init.d/puppet start #restart puppet
```

Documentation on fixing broken packages:
http://askubuntu.com/questions/58378/what-exactly-does-sudo-apt-get-f-install-do
http://askubuntu.com/questions/30993/fixing-software-center-catalog/183625#183625

**Comment by malachig** [comment]
I think I have this working now.  The above thread was becoming hard to follow so I have consolidated the instructions for setting up clia1 here:

https://github.com/genome/gms/blob/ubuntu-12.04/setup/system_tests/clia1

**Comment by sakoht** [comment]
Nice.

Is the /gscuser home dir required for things to run smoothly?

**Comment by malachig** [comment]
No just a matter of convenience for me so that I can configure my ssh 
keys and avoid annoying warnings when I log in.  This is totally 
specific to the clia1 which is not really a vanilla ubuntu install.

mg

**Comment by malachig** [comment]
Testing on clia1 is ongoing.  The latest test has been very smooth.  Almost through to the end of processing all HCC1395 data with no failures of any kind.

**Comment by malachig** [comment]
Testing on clia1 is not active right now.  At some point we will need to perform a major merge of improvements from `gms-core master` into `gms-core gms-pub`.  At that time we should also include the latest versions of each major cancer analysis pipeline.  When that happens we will need to perform extensive testing again and clia1 is the best place to do this.  However, for now I am closing this issue.
